pkg/virt-handler/vm.go

```diff
@@ -291,6 +292,12 @@ func (d *VMHandlerDispatch) processVmUpdate(vm *v1.VM, shouldDeleteVm bool) erro
 		return err
 	}
 
+	// Map Container Registry Disks to block devices Libvirt can consume
+	vm, err = registrydisk.MapRegistryDisks(vm)
+	if err != nil {
+		return err
+	}
```

Comment:
@berrange I have problems to see, why directly doing something, instead of hiding it behind a REST API, where we don't know if we need it, what it would do, or how it would look like, would introduce a technical dept. Completely independent if one likes a specific buzzword or not.

I agree with @davidvossel, that something like this PR opens possibilities for easy to integrate epemeral VMs without having to care about storage. If this proposed feature turns out, to not make sense, we can easily remove it again, if it makes sense, but we re-architect the storage part significantly later on, all the functional tests are still valuable ...

Also one note in general, regarding "microservices". So far KubeVirt, tries to follow as much as possible the Kubernetes architecture, which is very simple: As it's core, one binary for controllers, one binary for api-server extensions, and one binary for the node daemon. Only if it makes sense to exchange something arbitrarily, by third party implementations, they provide hooks or registration mechanisms for extension.

Here also a short summary, what people wanted to have extra containers/services/APIs or external service depenencies for withing the last two weeks, as part of the core infrastructure in KubeVirt:

 * Extra API and service for networking
 * Extra API and service for storage
 * A dependency on KubeDNS for our core infrastructure
 * Providing console stream access through a proxy (not through virt-api like right now), like spice/vnc
 * Having an extra credential resolver container for spice/vnc/console between the proxy and virt-api (which we don't have right now)

My intent is not to say that it is wrong to have one or all of them. But let's just write proposals for another two weeks, and say yes to all of them, then we have between 20 or 30 services.

Long story short: From my view, if it is something which can be done on the node, virt-handler should do it. Having extra daemons on the node, is not something I want to pursue (except for extensions/hooks where it makes sense of course).